sources/packages/backend/apps/api/src/services/student/student.service.ts

```diff
@@ -494,6 +494,78 @@ export class StudentService extends RecordDataModelService<Student> {
     return searchQuery.getMany();
   }
 
+  /**
+   * Search students based on the search criteria.
+   * @param searchCriteria options to search by firstName,
+   * lastName, appNumber or sin.
+   * @returns list of students.
+   */
+  async searchStudentApplicationForInstitution(
+    institutionId: number,
+    searchCriteria: {
+      firstName?: string;
+      lastName?: string;
+      appNumber?: string;
+      sin?: string;
+    },
+  ): Promise<Student[]> {
+    const searchQuery = this.repo
+      .createQueryBuilder("student")
+      .select([
+        "student.id",
+        "student.birthDate",
+        "user.firstName",
+        "user.lastName",
+        "sinValidation.sin",
+      ])
+      .innerJoin(
+        Application,
+        "application",
+        "application.student.id = student.id",
+      )
+
+      .innerJoin("application.location", "pirLocation")
+      .innerJoin("pirLocation.institution", "pirInstitution")
+
+      .leftJoin("application.currentAssessment", "studentAssessment")
+      .leftJoin("studentAssessment.offering", "offering")
+      .leftJoin("offering.institutionLocation", "offeringLocation")
+      .leftJoin("offeringLocation.institution", "offeringInstitution")
+
+      .innerJoin("student.user", "user")
+      .innerJoin("student.sinValidation", "sinValidation")
+      .where("user.isActive = true")
+      .andWhere(
+        "(offeringInstitution.id = :institutionId or pirInstitution.id = :institutionId)",
```

Comment:
Typeorm also has the "or" syntax.

Reply:
Yes, I see that but when I use .orWhere() it doesn't add a parenthesis to the query like in:
```sql
AND     ("offeringInstitution"."id" = $1   OR      "pirInstitution"."id"      = $2)
```
I'm not sure how to add those parenthesis through the DSL.

Reply:
Please see the below example:
```ts
   if (paginationOptions.searchCriteria) {
      activeApplicationQuery
        .andWhere(
          new Brackets((qb) => {
            qb.where(
              "CONCAT(user.firstName,' ', user.lastName) Ilike :searchCriteria",
            ).orWhere("application.applicationNumber Ilike :searchCriteria");
          }),
        )
        .setParameter(
          "searchCriteria",
          `%${paginationOptions.searchCriteria.trim()}%`,
        );
    }
```